frb_codegen/src/generator/dart/ty_sync_return.rs

```diff
@@ -26,22 +38,43 @@ impl TypeDartGeneratorTrait for TypeSyncReturnGenerator<'_> {
                         IrTypePrimitive::I64 => "Int64",
                         IrTypePrimitive::F32 => "Float32",
                         IrTypePrimitive::F64 => "Float64",
+                        IrTypePrimitive::Unit => {
+                            return "return;".to_owned();
+                        }
                         _ => panic!(
                             "SyncReturn generator for Dart: type {} is not supported",
                             primitive.rust_api_type()
                         ),
                     };
                     format!(
                         r#"
-                        final dataView = ByteData.view(raw.buffer);
-                        return dataView.get{primitive_name}(0);
-                        "#,
+                            final dataView = ByteData.view(raw.buffer);
+                            return dataView.get{primitive_name}(0);
+                            "#,
                         primitive_name = primitive_name
                     )
                 }
             },
             IrTypeSyncReturn::String => "return utf8.decode(raw);".into(),
             IrTypeSyncReturn::VecU8 => "return raw;".into(),
+            IrTypeSyncReturn::RustOpaque(o) => {
+                format!(
+                    "return {}.fromRaw(getOpaquePtr(raw), getOpaqueSize(raw), this);",
+                    o.inner_dart
+                )
+            }
+            IrTypeSyncReturn::DartOpaque(_) => {
+                "return _platform.inner.get_dart_object(getOpaquePtr(raw));".to_owned()
+            }
+            IrTypeSyncReturn::Option(_) => "".into(),
+        };
+        if let IrTypeSyncReturn::Option(ty) = &self.ir {
+            if let IrTypeSyncReturn::Option(_) = **ty {
+                panic!("Nested option is not suppored.")
+            }
```

Comment:
maybe move the panic up to that `match`? like:

```suggestion
            IrTypeSyncReturn::Option(_) => panic!("Nested option is not supported."),
        };
        if let IrTypeSyncReturn::Option(ty) = &self.ir {
```